Formula/consul-template.rb

```diff
@@ -16,50 +18,56 @@ class ConsulTemplate < Formula
 
   devel do
     url "https://github.com/hashicorp/consul-template.git",
-        :tag => "v0.18.0-rc1",
-        :revision => "933192dc0b6bb642af2549848e0681670550e095"
-    version "0.18.0-rc1"
-
-    # Upstream issue "Request: Makefile target that doesn't use docker"
-    # Reported 18 Nov 2016 https://github.com/hashicorp/consul-template/issues/793
-    patch :DATA
-  end
-
-  head do
-    url "https://github.com/hashicorp/consul-template.git"
-
-    patch :DATA
+        :tag => "v0.18.0-rc2",
+        :revision => "e446ab16fb567e1ddeb88ef29406b53bdb4ee387"
+    version "0.18.0-rc2"
   end
 
   depends_on "go" => :build
 
+  # This can be removed on next stable version
   go_resource "github.com/mitchellh/gox" do
     url "https://github.com/mitchellh/gox.git",
         :revision => "c9740af9c6574448fd48eb30a71f964014c7a837"
   end
 
   # gox dependency
+  # This can be removed on next stable version
   go_resource "github.com/mitchellh/iochan" do
     url "https://github.com/mitchellh/iochan.git",
         :revision => "87b45ffd0e9581375c491fef3d32130bb15c5bd7"
   end
 
   def install
-    ENV["GOPATH"] = buildpath
-
-    dir = buildpath/"src/github.com/hashicorp/consul-template"
-    dir.install buildpath.children
-
-    Language::Go.stage_deps resources, buildpath/"src"
-    ENV.prepend_create_path "PATH", buildpath/"bin"
-    cd("src/github.com/mitchellh/gox") { system "go", "install" }
-
-    cd dir do
-      system "make", "dev"
-      system "make", "test"
+    contents = Dir["{*,.git,.gitignore}"]
+    gopath = buildpath/"gopath"
+    (gopath/"src/github.com/hashicorp/consul-template").install contents
+
+    ENV["GOPATH"] = gopath
+    ENV.prepend_create_path "PATH", gopath/"bin"
+    arch = MacOS.prefer_64_bit? ? "amd64" : "386"
+    ENV["XC_OS"] = "darwin"
+    ENV["XC_ARCH"] = arch
+
+    cd gopath/"src/github.com/hashicorp/consul-template" do
+      # Ensure a clean git tree for version output
+      system "git", "reset", "--hard"
```

Comment:
I used this from the kubernetes-cli formula that also has a build time check for a clean branch. The problem detected here was `.brew_home` is in `buildpath.children`, so this is dirtying the branch. I've made a change to prevent `.brew_home` from being copied, and now the branch is reporting clean.